hibernate-core/src/test/java/org/hibernate/test/locking/OptimisticLockTest.java

```diff
@@ -0,0 +1,92 @@
+package org.hibernate.test.locking;
+
+import javax.persistence.Entity;
+import javax.persistence.Id;
+import javax.persistence.LockModeType;
+import javax.persistence.Version;
+
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.cfg.Configuration;
+
+import org.hibernate.testing.junit4.BaseCoreFunctionalTestCase;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class OptimisticLockTest extends BaseCoreFunctionalTestCase {
```

Comment:
I found there is already a similarly-named test class of `org.hibernate.test.optlock.OptimisticLockTest` already. If I didn't misunderstand, this test aims to solve the specific bug (jira annotation is needed?), so maybe we should name the testing class to better reflect its purpose? Maybe we could even move this class into the same package as the existing one?

Reply:
Thanks @NathanQingyangXu I changed the name of the test and moved it to the `o.h.test.optlock` package. I also changed the name of `version1` to `version`